pkgs/development/libraries/physics/hepmc3/default.nix

```diff
@@ -3,19 +3,21 @@
 let
   pythonVersion = with stdenv.lib.versions; "${major python.version}${minor python.version}";
   withPython = python != null;
+  # ensure that root is built with the same python interpreter, as it links against numpy
+  root_py = if withPython then root.override { inherit python; } else root;
```

Comment:
I don't think this makes a lot of sense to do this as HepMC doesn't interact with ROOT via any python API.

Reply:
it felt bad writing it, I'm just not aware of a different way to ensure that they both use the same interpreter version (if it's even important).

Reply:
Well it is an idiomatic code to ensure matching interpreter versions, but that is not needed in this particular case.

Reply:
right but just saying `root` doesn't give an indication what python interpreter will be used. As opposed to `python37Packages.hepmc3`

Reply:
I suppose HepMC3 links only to libRIO, which is not affected by python in any way. And `root`'s `setupHook` will set LD_LIBRARY_PATH, so hepmc might end up working with a different build of ROOT in some cases.